### ISSUE TITLE
Any reason why this issue was closed?  I'm having the same error.

### ISSUE THREAD

**Issue by Bambunote** [atom-community/markdown-preview-plus#527]
Any reason why this issue was closed?  I'm having the same error.

[Enter steps to reproduce:]

1. Update to 4.10.0
2. ...

**Atom**: 1.57.0 x64
**Electron**: 9.4.4
**OS**: macOS 11.4
**Thrown From**: [markdown-preview-plus](https://github.com/atom-community/markdown-preview-plus) package 4.10.0

### Stack Trace

Error reported when Markdown Preview Plus view is loading

```
At write EPIPE

Error: write EPIPE
    at afterWriteDispatched (internal/stream_base_commons.js:149:25)
    at writeGeneric (internal/stream_base_commons.js:140:3)
    at Socket._writeGeneric (net.js:776:11)
    at Socket._write (net.js:788:8)
    at doWrite (_stream_writable.js:435:12)
    at writeOrBuffer (_stream_writable.js:419:5)
    at Socket.Writable.write (_stream_writable.js:309:11)
    at /packages/markdown-preview-plus/dist/main.js:1:77939
    at new Promise (<anonymous>)
    at /packages/markdown-preview-plus/dist/main.js:1:77546)
    at /packages/markdown-preview-plus/dist/main.js:1:75833)
    at Ao.renderMarkdownText (/packages/markdown-preview-plus/dist/main.js:1:224277)
    at Ao.renderMarkdown (/packages/markdown-preview-plus/dist/main.js:1:223970)
```

### Commands

```
     -1:17.8.0 editor:consolidate-selections (input.hidden-input)
     -1:17.8.0 core:cancel (input.hidden-input)
     -1:16.8.0 core:select-up (input.hidden-input)
  4x -1:15.6.0 emmet:merge-lines (input.hidden-input)
  2x -1:10.4.0 core:move-left (input.hidden-input)
  3x -1:09.7.0 datatip:toggle (input.hidden-input)
     -1:07.3.0 core:backspace (input.hidden-input)
     -1:05.3.0 command-palette:toggle (input.hidden-input)
     -1:00.1.0 editor:consolidate-selections (input.hidden-input)
     -1:00.1.0 core:cancel (input.hidden-input)
     -0:58.9.0 emmet:merge-lines (input.hidden-input)
     -0:57.6.0 application:add-project-folder (input.hidden-input)
     -0:54.2.0 command-palette:toggle (input.hidden-input)
     -0:51.9.0 core:move-down (input.hidden-input)
     -0:51.4.0 core:confirm (input.hidden-input)
     -0:51.4.0 markdown-preview-plus:toggle (input.hidden-input)
```

### Non-Core Packages

```
atom-beautify 0.33.4 
atom-ide-debugger-python 0.7.3 
atom-ide-ui 0.13.0 
atom-language-r 1.4.8 
atom-live-server 2.3.0 
atom-ternjs 0.20.0 
autocomplete 0.47.0 
autocomplete-python 1.17.0 
autocomplete-sql 0.5.0 
busy-signal 2.0.1 
dash 1.7.1 
emmet 2.4.3 
git-blame 1.8.0 
git-time-machine 2.1.0 
go-debug 1.7.0 
go-plus 6.1.0 
go-signature-statusbar 1.2.3 
goto-definition 1.3.4 
Hydrogen 2.15.3 
hydrogen-launcher 1.2.2 
hydrogen-python 0.0.8 
ide-html 0.6.2 
ide-python 1.8.0 
ide-r 0.1.8 
ide-svelte 0.1.1 
indent-detective 0.4.0 
ink 0.12.6 
julia-client 0.12.6 
kite 0.197.0 
language-aadl 0.1.0 
language-julia 0.21.1 
language-latex 1.2.0 
language-matlab 0.2.1 
language-tex 0.4.0 
latex 0.50.2 
latex-completions 0.3.6 
linter 3.1.1 
linter-flake8 2.4.0 
markdown-pdf 2.3.3 
markdown-preview-plus 4.10.0 
pdf-view 0.72.0 
pen-paper-coffee-syntax 0.17.0 
pg-formatter 0.0.1 
prettier-atom 0.60.1 
python-debugger 0.2.0 
python-docstring 0.0.3 
react 0.18.0 
split-diff 1.4.0 
sync-settings 5.2.11 
tablr 1.8.3 
tikz-preview 0.2.8 
tool-bar 1.4.2 
uber-juno 0.3.0 
```

_Originally posted by @rjbergerud in https://github.com/atom-community/markdown-preview-plus/issues/508#issuecomment-858070591_

**Comment by Bambunote** [comment]
Sorry but I have the same problem.

**Comment by Bambunote** [comment]
My OS is Windows10 Professional

**Comment by Bambunote** [comment]
[Enter steps to reproduce:]

1. ...
2. ...

**Atom**: 1.57.0 x64
**Electron**: 9.4.4
**OS**: Unknown Windows version
**Thrown From**: [markdown-preview-plus](https://github.com/atom-community/markdown-preview-plus) package 4.11.0


### Stack Trace

Error reported when Markdown Preview Plus view is loading

```
At Cannot read property 'moveToSuccessor' of undefined

TypeError: Cannot read property 'moveToSuccessor' of undefined
    at HighlightIterator.moveToSuccessor (~/AppData/Local/atom/app-1.57.0/resources/app/static/<embedded>:11:3423669)
    at /packages/markdown-preview-plus/dist/main.js:1:199782
    at /packages/markdown-preview-plus/dist/main.js:1:198971
    at async Promise.all (index 15)
    at Ei (/packages/markdown-preview-plus/dist/main.js:1:197589)
    at /packages/markdown-preview-plus/dist/main.js:1:76242)
    at Bo.renderMarkdownText (/packages/markdown-preview-plus/dist/main.js:1:225684)
```

### Commands

```
     -4:31 tree-view:remove (span.name.icon.icon-file-text)
  2x -4:20.9.0 intentions:highlight (input.hidden-input)
     -4:18.3.0 markdown-preview-plus:toggle (input.hidden-input)
     -0:36.6.0 settings-view:open (a.btn-issue.btn.btn-error)
     -0:19.6.0 pane:reopen-closed-item (atom-workspace.workspace.scrollbars-visible-always.theme-one-dark-syntax.theme-one-dark-ui)
     -0:06.8.0 intentions:highlight (input.hidden-input)
     -0:06.0 markdown-preview-plus:toggle (input.hidden-input)
```

### Non-Core Packages

```
busy-signal 2.0.1 
gcc-make-run 0.2.12 
intentions 2.1.1 
linter 3.4.0 
linter-gcc 0.9.0 
linter-ui-default 3.4.1 
markdown-preview-plus 4.11.0 
simplified-chinese-menu 5.3.5 
```

**Comment by lierdakil** [comment]
Which is a completely different error... and Atom is being a donkey sending people to a closed issue. Investigating. May take a bit.

**Comment by lierdakil** [comment]
Actually, a minimal testcase would be really helpful, if you can provide one.

**Comment by lierdakil** [comment]
Yeah... actually, I can't really debug this without a testcase. Something apparently goes wrong in the Atom internals here, and without understanding what causes that, I can't really fix the issue. I only can suppress the error itself, which wouldn't be very helpful. So I really need a document on which this happens and maybe a reproduction guide if this doesn't happen every time.

In the meantime, the error is thrown from tree-sitter highlighter, apparently. The workaround is to switch to the legacy highlighter, i.e. set this in the markdown-preview-plus settings:
![showing "Code Highlighter" option set to "legacy"](https://user-images.githubusercontent.com/7275622/127149912-9c01f7c0-37fc-4f4b-ab1c-69c697d1e5e4.png)

**Comment by Bambunote** [comment]
SORRY but I am just a noob user ( a middle school student from China ) . Please forgive me for not answering your comment. I was busy fixing a bug in another code of mine. There was an update uploaded and I just downloaded it and it may (or must) fix the bug. Thanks Atom (Though I've just used it for only 5 days) . Maybe I can close the comment. Thanks @lierdakil .

**Comment by Bambunote** [comment]
Sorry Maybe I can close the issue.

**Comment by Bambunote** [comment]
Thank you , but it does disappeared.


Maybe the update also fixed the problem indirectly.




------------------&nbsp;原始邮件&nbsp;------------------
发件人: "Nikolay ***@***.***&gt;; 
发送时间: 2021年7月27日(星期二) 晚上10:06
收件人: ***@***.***&gt;; 
抄送: ***@***.***&gt;; "State ***@***.***&gt;; 
主题: Re: [atom-community/markdown-preview-plus] Any reason why this issue was closed?  I&#39;m having the same error. (#527)





 
The point of the update was to fix the hard crash only, but the error message should still pop up (if it happens, that is). Anyway, if the issue doesn't reproduce, then maybe it was a one-off hiccup. If a scary-looking error message pops up again, though, feel free to open a new issue. Also, when creating an issue here, try to include sample input (i.e. Markdown source) which prompts the error message if possible and/or applicable.
